errdefs/defs.go

```diff
@@ -52,3 +57,18 @@ type ErrNotImplemented interface {
 type ErrUnknown interface {
 	Unknown()
 }
+
+// ErrCancelled signals that the action was cancelled.
+type ErrCancelled interface {
```

Comment:
I notice there's no HTTP translation for `ErrCancelled`, `ErrDeadline` and `ErrDataLoss` in httputils/errors.go, correct?

Reply:
Correct, it's just a 500.

Reply:
nit: perhaps make it explicit, so that it's clear it's a deliberate choice to make them 500's, and not because we "forgot"

Reply:
Oh, never mind; didn't look at the code yet; I see you added them :heart: